src/Contexts/User-Context/user-context.js

```diff
@@ -0,0 +1,24 @@
+import { createContext, useReducer, useContext, useState } from "react";
+import { initialUserState, userReducer } from "./user-reducer";
+import { capitalize } from "./user-utils";
+const UserContext = createContext(initialUserState);
+
+const UserProvider = ({ children }) => {
+  const [userState, userDispatch] = useReducer(userReducer, initialUserState);
+  const [alert, setAlert] = useState({ type: "", message: "" });
```

Comment:
If you are using usereducer then you can add one more state in reducer itself instead of taking usestate

Reply:
made changes